AWSKinesis/AWSAbstractKinesisRecorder.m

```diff
@@ -194,7 +203,8 @@ - (AWSTask *)saveRecord:(NSData *)data
             [self.recorderHelper checkByteThresholdForNotification:notificationByteThreshold
                                                 notificationSender:notificationSender
                                                           fileSize:fileSize];
-            if (fileSize > diskByteLimit) {
+            if ((diskByteLimit > 0) &&
```

Comment:
Please add a check for `AWSKinesisAbstractClientBatchRecordCountLimitMax` as well.